mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -868,12 +871,23 @@ fn try_parse_generic_error(request_result: &MetaRequestResult) -> Option<S3Reque
         }
     }
 
+    /// Try to look for error related to no signing credentials
+    fn try_parse_no_credentials(request_result: &MetaRequestResult) -> Option<S3RequestError> {
+        let crt_error_code = request_result.crt_error.raw_error();
+        if crt_error_code == 6146 {
+            Some(S3RequestError::NoSigningCredentials)
+        } else {
+            None
```

Comment:
I think these cases are all CRT errors because normal http status shouldn't be zero? if that's true we can return `CrtError` here.

Reply:
But, if they are `None`, it will still return `ObjectClientError::ClientError(S3RequestError::ResponseError(request_result)))` in the `make_meta_request_from_options` method. So, I am not sure whether we should return `CrtError` here.

Reply:
I think that will turn into `Unknown response error` when being printed out to the console, it's more meaningful to return `CrtError` when we already know that.